python_modules/dagster/dagster/_core/definitions/multi_dimensional_partitions.py

```diff
@@ -271,6 +273,24 @@ def has_partition_key(
                 return False
         return True
 
+    @lru_cache(maxsize=1)
+    def _get_partition_keys(
```

Comment:
it's quite expensive to get all the partition keys for a massive multi-asset, so let's try not to do that more than once if we can avoid it.

Reply:
Can we use `cached_method` here?

Reply:
Otherwise if there are multiple partitions_defs in play I think this will get thrashed?

Reply:
bumped to maxsize=5, but the reason I didn't go with cached_method is that it has infinite size. these sets of partitions can be pretty massive, so I wanted to put some guardrail on the total number of things we'd keep cached in a long-living process.